cfme/configure/access_control.py

```diff
@@ -403,77 +636,154 @@ def __init__(self, name=None, vm_restriction=None, product_features=None, applia
         self.product_features = product_features or []
 
     def create(self):
-        navigate_to(self, 'Add')
-        fill(self.form, {'name_txt': self.name,
-                         'vm_restriction_select': self.vm_restriction,
-                         'product_features_tree': self.product_features},
-             action=form_buttons.add)
-        flash.assert_success_message('Role "{}" was saved'.format(self.name))
+        view = navigate_to(self, 'Add')
+        view.fill({'name_txt': self.name,
+                   'vm_restriction_select': self.vm_restriction,
+                   'product_features_tree': self.product_features})
+        view.add_button.click()
+        view = self.create_view(AllRolesView)
+        view.flash.assert_success_message('Role "{}" was saved'.format(self.name))
 
     def update(self, updates):
-        navigate_to(self, 'Edit')
-        fill(self.form, {'name_txt': updates.get('name'),
-                         'vm_restriction_select': updates.get('vm_restriction'),
-                         'product_features_tree': updates.get('product_features')},
-             action=form_buttons.save)
-        flash.assert_success_message('Role "{}" was saved'.format(updates.get('name', self.name)))
+        view = navigate_to(self, 'Edit')
+        view.fill({'name_txt': updates.get('name'),
```

Comment:
You don't have to put whole dictionary here, `view.fill(updates)` will be enough.